### PR TITLE
Fix WeightedLeastSquaresTest for GSL2

### LINKED ISSUE #16679: Fix WeightedLeastSquaresTest for GSL2
Tests failing with GSL2: http://builds.mantidproject.org/job/master_systemtests-ubuntu-16.04/32/testReport/SystemTests/WeightedLeastSquaresTest/WeightedLeastSquaresTest/
Fix:
- SystemTests.WeightedLeastSquaresTest.WeightedLeastSquaresTest

This system test may also need revisiting now that we have more fitting tests in place.

### PR DESCRIPTION
**Pull request by mantid-roman**
Removed a test case which shouldn't be there in the first place. Increased tolerance for another test.

**To test:**

System tests must pass.

Fixes #16679.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Testing/SystemTests/tests/analysis/WeightedLeastSquaresTest.py [hunk 1]
```diff
@@ -118,27 +118,6 @@ def setUp(self):
         if not self.__class__.workspace:
             self.__class__.workspace = load_fitting_test_file_ascii(self.filename)
 
-    def test_misses_expected_solution(self):
```
Comment (martyngigg) [inline Testing/SystemTests/tests/analysis/WeightedLeastSquaresTest.py]: What's the reason this test shouldn't exist?

Comment (mantid-roman) [inline Testing/SystemTests/tests/analysis/WeightedLeastSquaresTest.py]: The fit doesn't converge on all platforms and gives a result that nobody would want to use for anything.

Comment (martyngigg) [inline Testing/SystemTests/tests/analysis/WeightedLeastSquaresTest.py]: I see. Thanks for clarifying that.

**Review by martyngigg — COMMENTED** [review]
I'm happy with the code changes here.

**Review by martyngigg — APPROVED** [review]
:shipit:

**Pull request merged**